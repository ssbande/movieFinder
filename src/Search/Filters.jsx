import React from 'react';
import PropTypes from 'prop-types';
import { Form } from 'semantic-ui-react';

import constants from '../utils/constants';
import { FilterProps } from '../utils/propTypes';

const Filters = ({ filter, handleFilterChange }) => {
  const refFilters = Object.values(constants.config);

  return (
    <Form.Group inline id='filterForm'>
      <label>Search for </label>
      {refFilters.sort((a, b) => a.order - b.order).map(fltr => {
        const { label, category, order } = fltr;
        return <Form.Radio key={order}
          label={label}
          value={category}
          checked={filter === category}
          onChange={handleFilterChange}
        />
      })}
    </Form.Group>
  )
}

Filters.propTypes = {
  filter: FilterProps,
  handleFilterChange: PropTypes.func.isRequired
}

export default Filters;