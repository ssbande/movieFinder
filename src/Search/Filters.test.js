import React from 'react';
import { shallow } from 'enzyme';

import Filters from './Filters';

const setup = () => ({
  props: {
    filter: 'all',
    handleFilterChange: jest.fn(),
  }
})

describe('All Filter options for the app', () => {
  it('should show all filter options correctly', () => {
    const { props } = setup();
    const wrapper = shallow(<Filters {...props} />);
    expect(wrapper).toMatchSnapshot();
  })
})