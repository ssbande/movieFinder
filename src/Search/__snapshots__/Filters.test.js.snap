// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Filter options for the app should show all filter options correctly 1`] = `ShallowWrapper {}`;
